src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -95,85 +94,51 @@ private BoundExpression BindWithExpression(WithExpressionSyntax syntax, Diagnost
-                            {
-                                case SymbolKind.Property:
-                                    member = sym;
-                                    // PROTOTYPE: this should check for init-only, but that isn't a separate feature yet
-                                    // It also will not work in metadata.
-                                    if (!(sym is SynthesizedRecordPropertySymbol))
-                                    {
-                                        goto default;
-                                    }
-                                    break;
-
-                                default:
-                                    hasErrors = true;
-                                    diagnostics.Add(
-                                        ErrorCode.ERR_WithMemberIsNotRecordProperty,
-                                        location);
-                                    break;
-                            }
-                        }
-
-                        if (!hasErrors && member is null)
-                        {
-                            hasErrors = true;
-                            Error(
-                                diagnostics,
-                                ErrorCode.ERR_NoSuchMemberOrExtension,
-                                location,
-                                cloneReturnType,
-                                propName);
-                        }
-                    }
-
-                    boundRight = BindValue(assignment.Right, diagnostics, BindValueKind.RValue);
-                    if (!(member is null))
-                    {
-                        boundRight = GenerateConversionForAssignment(
-                            member.GetTypeOrReturnType().Type,
-                            boundRight,
-                            diagnostics);
-                    }
-                    lookupResult.Clear();
+                    BoundExpression? boundMember = null;
+                    boundMember = BindInstanceMemberAccess(
+                        node: left,
+                        right: left,
+                        boundLeft: implicitReceiver,
+                        rightName: propName,
+                        rightArity: 0,
+                        typeArgumentsSyntax: default(SeparatedSyntaxList<TypeSyntax>),
+                        typeArgumentsWithAnnotations: default(ImmutableArray<TypeWithAnnotations>),
+                        invoked: false,
+                        indexed: false,
+                        diagnostics: diagnostics);
+
+                    hasErrors |= boundMember.HasAnyErrors || implicitReceiver.HasAnyErrors;
+
+                    boundMember = CheckValue(boundMember, BindValueKind.Assignable, diagnostics);
+
+                    var boundRight = BindValue(assignment.Right, diagnostics, BindValueKind.RValue);
+                    boundExpr = BindAssignment(expr, boundMember, boundRight, isRef: false, diagnostics);
```

Comment:
>isRef: false, [](start = 78, length = 13)

Could we get in a situation with a ref property and a ref value? Please add a test

```
class C 
{
    ref int Property => throw null;
    
    public void M(C c) 
    {
        new C() { Property = c.Property };
        
        _ = c with { Property = c.Property };
    }
}
```